### PR TITLE
Added search on clans page

### PR DESCRIPTION
**Pull request by Paul-Vasylenko**
/clans?search=:clanName 

https://trello.com/c/7UfJ0tvV/159-clans-add-searchclanname-param-for-emails

### CODE REVIEW

**Comment by Paul-Vasylenko** [comment]
[[Clans] Add ?search=:clanName param for emails](https://trello.com/c/7UfJ0tvV/159-clans-add-searchclanname-param-for-emails)

File: client/src/containers/clans/clan/logic/saga.ts [hunk 1]
```diff
@@ -31,8 +34,17 @@ export function* leaveClanWorker(action: ReturnType<typeof actions.leaveClan>):
...
+			setNotificationState({
+				state: {
+					notificationType: NotificationType.Error,
+					message: response.message,
+				},
+			}),
+		);
 	} else {
 		yield put(userActions.setUser({ user: response.user }));
+		historyHelper.push('/clans');
```
Comment (svitlua) [inline client/src/containers/clans/clan/logic/saga.ts:47]: use ROUTES const here

File: client/src/containers/clans/clans/index.tsx [hunk 2]
```diff
@@ -35,7 +48,24 @@ const Clans: React.FC = () => {
 	const leaveClan = (id: string) => {
 		dispatch(actions.leaveClan({ id }));
 	};
-
+	const handleGoToClan = () => {
+		if (user?.clan !== null) {
+			historyHelper.push('/clan');
```
Comment (svitlua) [inline client/src/containers/clans/clans/index.tsx:53]: use ROUTES const

File: client/src/containers/clans/clans/index.tsx [hunk 3]
```diff
@@ -35,7 +48,24 @@ const Clans: React.FC = () => {
...
+						notificationType: NotificationType.Error,
+						message: 'You don`t have a clan',
+					},
+				}),
+			);
+		}
+	};
+	const handleGoToClanAllClans = () => {
+		setIsSearch(false);
+		historyHelper.push('/clans');
```
Comment (svitlua) [inline client/src/containers/clans/clans/index.tsx:67]: the same

**Review by svitlua — CHANGES_REQUESTED** [review]
I don't really understand why we need to have this functionality **/clans?search=:clanName** 
There is already possibility to search by name on Clans page
image


I think what we really need is to refactor Clan page to have something like **clan/:clanname** route

Please, explain

**Comment by Paul-Vasylenko** [comment]
useless

**Pull request closed without merging**